pkg/controller/controller_binding.go

```diff
@@ -577,108 +584,204 @@ func (c *controller) reconcileServiceBinding(binding *v1beta1.ServiceBinding) er
+		glog.Warning(pcb.Message(s))
+		c.recorder.Eventf(binding, corev1.EventTypeWarning, errorEjectingBindReason, "%v %v", errorEjectingBindMessage, s)
+		setServiceBindingCondition(
+			toUpdate,
+			v1beta1.ServiceBindingConditionReady,
+			v1beta1.ConditionUnknown,
+			errorEjectingBindReason,
+			errorEjectingBindMessage+s,
+		)
+		if _, err := c.updateServiceBindingStatus(toUpdate); err != nil {
 			return err
 		}
+		return err
+	}
 
-		unbindRequest := &osb.UnbindRequest{
-			BindingID:  binding.Spec.ExternalID,
-			InstanceID: instance.Spec.ExternalID,
-			ServiceID:  serviceClass.Spec.ExternalID,
-			PlanID:     servicePlan.Spec.ExternalID,
+	if toUpdate.DeletionTimestamp == nil {
+		if toUpdate.Status.OperationStartTime == nil {
+			now := metav1.Now()
+			toUpdate.Status.OperationStartTime = &now
 		}
+	} else {
+		if toUpdate.Status.CurrentOperation != v1beta1.ServiceBindingOperationUnbind {
+			// Cancel any pending orphan mitigation since the resource is being deleted
+			toUpdate.Status.OrphanMitigationInProgress = false
 
-		if utilfeature.DefaultFeatureGate.Enabled(scfeatures.OriginatingIdentity) {
-			originatingIdentity, err := buildOriginatingIdentity(binding.Spec.UserInfo)
+			toUpdate, err = c.recordStartOfServiceBindingOperation(toUpdate, v1beta1.ServiceBindingOperationUnbind)
 			if err != nil {
-				s := fmt.Sprintf(`Error building originating identity headers while unbinding: %v`, err)
-				glog.Warning(pcb.Message(s))
-				c.recorder.Event(binding, corev1.EventTypeWarning, errorWithOriginatingIdentity, s)
-				setServiceBindingCondition(
-					toUpdate,
-					v1beta1.ServiceBindingConditionReady,
-					v1beta1.ConditionFalse,
-					errorWithOriginatingIdentity,
-					s,
-				)
-				if _, err := c.updateServiceBindingStatus(toUpdate); err != nil {
-					return err
-				}
+				// There has been an update to the binding. Start reconciliation
+				// over with a fresh view of the binding.
 				return err
 			}
-			unbindRequest.OriginatingIdentity = originatingIdentity
 		}
+	}
 
-		if toUpdate.DeletionTimestamp == nil {
-			if toUpdate.Status.OperationStartTime == nil {
-				now := metav1.Now()
-				toUpdate.Status.OperationStartTime = &now
-			}
-		} else {
-			if toUpdate.Status.CurrentOperation != v1beta1.ServiceBindingOperationUnbind {
-				// Cancel any pending orphan mitigation since the resource is being deleted
-				toUpdate.Status.OrphanMitigationInProgress = false
-
-				toUpdate, err = c.recordStartOfServiceBindingOperation(toUpdate, v1beta1.ServiceBindingOperationUnbind)
-				if err != nil {
-					// There has been an update to the binding. Start reconciliation
-					// over with a fresh view of the binding.
-					return err
-				}
-			}
+	if binding.Status.UnbindStatus == v1beta1.ServiceBindingUnbindStatusRequired {
+		if ok, err := c.serviceBindingRequestUnbinding(binding, toUpdate, pcb); !ok || err != nil {
+			return err
 		}
+	}
+
+	if toUpdate.Status.OrphanMitigationInProgress {
```

Comment:
The Controller is doing a number of things here and earlier before the call to `serviceBindingRequestUnbinding` that are not necessary if the `UnbindStatus` is not `Required`. For example, the Controller should not be setting the Ready condition, particularly with a `successUnboundReason`. All the Controller should do is delete the finalizers.